applications/tari_validator_node/src/p2p/rpc/service_impl.rs

```diff
@@ -290,6 +292,57 @@ where TPeerProvider: PeerProvider + Clone + Send + Sync + 'static
         Ok(Response::new(resp))
     }
 
+    async fn sync_blocks(&self, req: Request<BlockSyncRequest>) -> Result<Streaming<BlockSyncResponse>, RpcStatus> {
+        let (sender, receiver) = mpsc::channel(100);
+        let req = req.into_message();
+        let shard_db = self.shard_state_store.clone();
+
+        let start_block_id = BlockId::try_from(req.start_block_id)
+            .map_err(|e| RpcStatus::bad_request(&format!("Invalid encoded block id: {}", e)))?;
+        // Check if we have the blocks
+        shard_db
+            .with_read_tx(|tx| Block::get(tx, &start_block_id))
+            .map_err(|e| RpcStatus::bad_request(&e))?;
+        let epoch = req.epoch;
+        // Sync from tip, or from the block that the requester has.
+        let mut block = match req.end_block_id.len() {
+            0 => shard_db.with_read_tx(|tx| Block::get_tip(tx, tari_dan_common_types::Epoch(epoch))),
```

Comment:
You'll have to change get_tip to return the MAX height and to not take in an epoch anymore. It changed with the leader failure PR not to have a chain per epoch but I don't think this function was changed.